src/canister/individual_user_template/src/api/cdao/airdrop.rs

```diff
@@ -0,0 +1,108 @@
+use candid::{Nat, Principal};
+use ic_base_types::PrincipalId;
+use ic_cdk_macros::update;
+use ic_sns_root::pb::v1::{ListSnsCanistersRequest, ListSnsCanistersResponse};
+use icrc_ledger_types::icrc1::{account::Account, transfer::{Memo, TransferArg, TransferError}};
+use shared_utils::canister_specific::individual_user_template::types::{error::AirdropError, profile::UserProfileDetailsForFrontendV2};
+
+use crate::CANISTER_DATA;
+
+#[update]
+async fn request_airdrop(token_root: Principal, memo: Option<Memo>, amount: Nat, user_canister: Principal) -> Result<(), AirdropError> {
+    let current_caller = ic_cdk::caller();
+    let profile_info = get_profile_info(user_canister).await?;
+    
+    if profile_info.principal_id != current_caller {
+        return Err(AirdropError::CanisterPrincipalDoNotMatch);
```

Comment:
We need to add a check if the `user_canister` is from our network of Yral canisters. We can add a TODO comment for now as the changes to support this are still in progress.

Reply:
Sure